pkg/frontend/middleware/validate.go

```diff
@@ -23,100 +22,103 @@ const (
-					return
-				}
+		if _, found := vars["resourceType"]; found {
+			if vars["resourceType"] != strings.ToLower(resourceType) {
+				api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceType, "", "The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["resourceType"], vars["resourceProviderNamespace"], apiVersion)
+				return
 			}
+		}
 
-			if _, found := vars["location"]; found {
-				if !strings.EqualFold(vars["location"], env.Location()) {
-					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidLocation, "", "The provided location '%s' is malformed or invalid.", vars["location"])
-					return
-				}
+		if _, found := vars["resourceName"]; found {
+			if !rxResourceGroupName.MatchString(vars["resourceName"]) {
+				api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s/%s' under resource group '%s' was not found.", vars["resourceProviderNamespace"], vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
+				return
 			}
+		}
 
-			if _, found := vars["operationId"]; found {
-				valid := uuid.IsValid(vars["operationId"])
-				if !valid {
-					api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidOperationID, "", "The provided operation identifier '%s' is malformed or invalid.", vars["operationId"])
-					return
-				}
+		if _, found := vars["location"]; found {
+			if !strings.EqualFold(vars["location"], v.Location) {
+				api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidLocation, "", "The provided location '%s' is malformed or invalid.", vars["location"])
+				return
 			}
+		}
 
-			queries, err := route.GetQueriesTemplates()
-			var hasVariableAPIVersion bool
-			for _, query := range queries {
-				if query == "api-version=" {
-					hasVariableAPIVersion = true
-					break
-				}
+		if _, found := vars["operationId"]; found {
+			valid := uuid.IsValid(vars["operationId"])
+			if !valid {
+				api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidOperationID, "", "The provided operation identifier '%s' is malformed or invalid.", vars["operationId"])
+				return
+			}
+		}
+
+		queries, err := route.GetQueriesTemplates()
+		var hasVariableAPIVersion bool
+		for _, query := range queries {
+			if query == "api-version=" {
+				hasVariableAPIVersion = true
+				break
 			}
+		}
 
-			if err != nil || hasVariableAPIVersion {
-				if apiVersion != "" {
-					if _, found := apis[apiVersion]; !found {
-						api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceType, "", "The resource type '%s' could not be found in the namespace '%s' for api version '%s'.", vars["resourceType"], vars["resourceProviderNamespace"], apiVersion)
-						return
-					}
+		if err != nil || hasVariableAPIVersion {
+			if apiVersion != "" {
```

Comment:
nit: can we reduce this nesting at least one level by combining 113 and 114
`(err != nil || hasVariableAPIVersion) && apiVersion != ""`

Reply:
wasn't the main purpose of that pr but yeah that's a simple enough change that doesn't blow up the scope. will do

Reply:
ah, I thought part of the goal was to reduce nesting, thanks for including this change.

Reply:
so the point was to remove `a function that returns a function that returns a function` and go back to the standard middleware pattern

Reply:
and also my bad for not having a nice description